srcCpp/perftestReadersListeners.h

```diff
@@ -0,0 +1,146 @@
+  #include <unistd.h>
+  #include <signal.h>
+#endif
+
+/*********************************************************
+ * Listener for the Subscriber side
+ *
+ * Keeps stats on data received per second.
+ * Returns a ping for latency packets
+ */
+class ThroughputListener : public IMessagingCB {
+public:
+    unsigned long long packetsReceived;
+    unsigned long long bytesReceived;
+    unsigned long long missingPackets;
+    int lastDataLength;
+    bool changeSize;
+
+    // store info for the last data set
+    int intervalDataLength;
+    unsigned long long intervalPacketsReceived;
+    unsigned long long intervalBytesReceived;
+    unsigned long long intervalMissingPackets;
+    unsigned long long intervalTime, beginTime;
+    float missingPacketsPercent;
+    CpuMonitor cpu;
+
+private:
+    IMessagingReader *_reader;
+    IMessagingWriter *_writer;
+    unsigned long *_lastSeqNum;
+
+    int _numPublishers;
+    std::vector<int> _finishedPublishers;
+    bool _useCft;
+
+public:
+    ThroughputListener(
+            IMessagingWriter *writer,
+            IMessagingReader *reader = NULL,
+            bool UseCft = false,
+            int numPublishers = 1);
+
+    ~ThroughputListener();
+
+    IMessagingReader *get_reader();
+
+    void ProcessMessage(TestMessage &message);
+
+    void print_summary(TestMessage &message);
+
+
+};
+
+
+/*********************************************************
+ * Data listener for the Announcement
+ *
+ * Receives an announcement message from a Subscriber once
+ * the subscriber has discovered every Publisher.
+ */
+class AnnouncementListener : public IMessagingCB {
+public:
+    std::vector<int> subscriberList;
+
+    AnnouncementListener(){}
+
+    void ProcessMessage(TestMessage &message);
+};
+
+/*********************************************************
+ * Data listener for the Publisher side.
+ *
+ * Receives latency ping from Subscriber and does
+ * round trip latency calculations
+ */
+class LatencyListener : public IMessagingCB {
+private:
+    unsigned long long _latencySum;
+    unsigned long long _latencySumSquare;
+    unsigned long long _count;
+    unsigned long _latencyMin;
+    unsigned long _latencyMax;
+    int _lastDataLength;
+    unsigned long *_latencyHistory;
+    unsigned long _clockSkewCount;
+    unsigned int _numLatency;
+    IMessagingWriter *_writer;
+    IMessagingReader *_reader;
+
+public:
+    CpuMonitor cpu;
+
+public:
+    LatencyListener(
+            unsigned int numLatency,
+            IMessagingReader *reader,
+            IMessagingWriter *writer);
+
+    ~LatencyListener();
+
+    IMessagingReader *get_reader();
+
+    void print_summary_latency();
+
+    void ProcessMessage(TestMessage &message);
+};
+
+
+/*
+ * perftestReaderListener.cxx have some dependencies from perftest_cpp.h.
+ * Also perftest_cpp.h needs the tree classes defined here.
+ * As solution, the include of perftest_cpp.h has been moved to the
+ * end of this file.
+ */
+#include "perftest_cpp.h"
```

Comment:
Actually, I would like to know what do you think about the include that I need to add at the end of the file. @alexcamposruiz 

Other solution is place it on the .cxx file.